src/Lachain.Core/RPC/HTTP/Web3/TransactionServiceWeb3.cs

```diff
@@ -572,7 +576,25 @@ public string Call(JObject opts, string? blockId)
         [JsonRpcMethod("eth_gasPrice")]
         public string GetNetworkGasPrice()
         {
-            return Web3DataFormatUtils.Web3Number(_stateManager.CurrentSnapshot.NetworkGasPrice.ToUInt256());
+            var gasLimit = GasMetering.DefaultBlockGasLimit;
+
+            var latestBlock = _blockManager.LatestBlock();
+            var latestBlockHash = latestBlock.Hash.ToHex();
+            var lastBlockGasUsed = GetGasUsed(latestBlockHash);
+
+            Logger.LogInformation($"lastBlockGasUsed:: {lastBlockGasUsed}");
+            Logger.LogInformation($"latestBlockHash:: {latestBlockHash}");
+
+            if (lastBlockGasUsed > gasLimit / 10)
```

Comment:
consider adding !(lastBlockGasUsed is null) && lastBlockGasUsed > gasLimit / 10

Reply:
I think it is better return 0 instead of null in GetGasUsed